app/models/petition.rb

```diff
@@ -1082,4 +1084,20 @@ def state_for_publishing(time)
   def time_for_publishing(time)
     open_at || time
   end
+
+  def set_threshold_for_referral
+    self.threshold_for_referral = Site.threshold_for_referral
```

Comment:
If we use `||=` then we can pass the value when building the instance (e.g. using a FactoryBot build or create).

Reply:
Adding `||=` here seems to result in the attribute actually being saved as `nil`!

I _think_ this is because I've overwritten `#threshold_for_referral` to return the site value if the attribute is nil. 

So `self.threshold_for_referral` here doesn't return `nil`, meaning that the attribute is not set 🤯

Reply:
Ah, of course - you should be able to override the accessor like this:

``` ruby
self[:threshold_for_referral] ||= Site.threshold_for_referral
```

Reply:
Oh yes! Nice 😀